### PR TITLE
Task 7 - vote on an article

### PR DESCRIPTION
**Pull request by gilaineyo**

### CODE REVIEW

File: src/components/Articles/ArticleFocus/ArticleFocus.jsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    const { title, author, body, created_at, comment_count, topic, article_img_url } = currArticle
+    const postedAt = new Date(created_at)
+
+    const handleVote = () => {
+        const vote = { inc_votes: !hasVoted ? 1 : -1}
+        setLocalVotes(localVotes + vote.inc_votes)
+        patchArticle(article_id, vote)
+        .then((article) => {
+            setLocalVotes(article.votes)
+            setHasVoted(!hasVoted)
```
Comment (nc-nick-jones) [inline src/components/Articles/ArticleFocus/ArticleFocus.jsx]: nice disabling of the button!

File: src/components/Articles/ArticleFocus/ArticleFocus.jsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        })
+        .catch((err) => {
+            if (hasVoted) {
+                setLocalVotes(localVotes - 1)
+            } else {
+                setLocalVotes(localVotes + 1)
+            }
+            setHasVoted(!hasVoted)
+        })
```
Comment (nc-nick-jones) [inline src/components/Articles/ArticleFocus/ArticleFocus.jsx:41]: I am wondering why all this is in the catch block?

**Review by nc-nick-jones — CHANGES_REQUESTED** [review]
Great work Gilaine.  Would you be able to add some user feedback based on patchVotes failing?  If we're offline currently it appears as though our votes were successful. 
Other than that,  all good!   
Also noticed you have some merge conflict.  Could you address these and the above feedback point and resubmit?  Thanks!

File: src/components/Articles/ArticleFocus/ArticleFocus.jsx (hunk 2 above)
Comment (gilaineyo) [reply src/components/Articles/ArticleFocus/ArticleFocus.jsx:41]: My idea was that it's optimistically rendered until the results of the patch and if the patch fails it puts the button back to normal (voted/unvoted depending on what the user tried to do). I just broke the patch code and can see now that it's not correcting its previous optimism and the catch block isn't doing anything but I don't know why. I'm just really stuck on this.

Comment (nc-nick-jones) [reply src/components/Articles/ArticleFocus/ArticleFocus.jsx:41]: Hi Gilaine,  that was the right way to think about it -  add a catch block to the promise chain in your handle vote function, and inside it decrement the value of votes and set some state that will enable an error message to be shown to the user.  If the util function returned an error this catch block would be entered.  
I was just confused by the 'hasVoted' condition being dealt with in the catch as well.   
Your approach here is effectively doing the same thing, just relying on a particular return value from the util function if an error is returned.

**Review by nc-nick-jones — APPROVED** [review]
Hi Gilaine this is good! See comment for details.   I would suggest that  you add some kind of error message for user feedback, when vote does not go through.   
Also see my reply to your comment for details on how to do this differently - when you have a chance in the future, give refactoring a go.  Nice work.

**Pull request merged**